### PR TITLE
Migrate to user-centric config name for `cargo check` stuff

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

**Comment by matklad** [comment]
cc @edwin0cheng

File: crates/rust-analyzer/src/config.rs [hunk 1]
```diff
@@ -46,9 +46,9 @@ config_data! {
         cargo_allFeatures: bool          = "false",
         /// List of features to activate.
         cargo_features: Vec<String>      = "[]",
-        /// Run `cargo check` on startup to get the correct value for package
-        /// OUT_DIRs.
-        cargo_loadOutDirsFromCheck: bool = "false",
+        /// Run build scripts (`build.rs`) for more precise code analysis.
+        cargo_runBuildScripts |
+        cargo_loadOutDirsFromCheck : bool = "false",
```
Comment (lnicola) [inline crates/rust-analyzer/src/config.rs:51]:
```suggestion
        cargo_loadOutDirsFromCheck: bool = "false",
```

**Pull request merged**